File: fapi/src/main/java/com/chernyak/fapi/models/Project.java
package com.chernyak.fapi.models;

public class Project {

    private Long id;
    private String projectCode;
    private String name;
    private String summary;
    private String manager;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getProjectCode() {
        return projectCode;
    }

    public void setProjectCode(String projectCode) {
        this.projectCode = projectCode;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

    public String getManager() {
        return manager;
    }

    public void setManager(String manager) {
        this.manager = manager;
    }
}

File: frontend/src/app/modules/project/models/Project.ts
export class Project {  
  id: string;
  code: string;  
  name: string;
  summary: string; 
  manager: string;

  static cloneBase(project: Project): Project {
    const clonedProject: Project = new Project();
    clonedProject.id = project.id;
    clonedProject.code = project.code;
    clonedProject.name = project.name;
    clonedProject.summary = project.summary;
    clonedProject.manager = project.manager;    
    return clonedProject;
  }
}